### ISSUE TITLE
inconsistent-return-statements not working in try branches

### ISSUE THREAD

**Issue by joshuahaertel** [PyCQA/pylint#3468]
### Steps to reproduce
Add a return in the try clause
```
def foo(bar):
    try:
        return bar.baz
    except AttributeError:
        pass
```

### Current behavior
Pylint emits no warning for `inconsistent-return-statements`. Note that it will emit a warning if the value is returned in the except clause.

### Expected behavior
Pylint should emit a warning for not explicitly returning a value if an exception is hit, be it in the except clause itself or at the end of the call.

### pylint --version output
```
$ pylint --version output
pylint 2.4.4
astroid 2.3.3
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0]
```

**Comment by joshuahaertel** [comment]
Also worth noting that returning in some except clauses and not in others does not emit a warning either.
```
def foo(bar):
    try:
        return bar.baz
    except AttributeError:
        pass
    except KeyError:
        return True
```

**Comment by PCManticore** [comment]
@joshuahaertel I can't quite reproduce this using:
```
pylint 2.4.4
astroid 2.3.3
Python 3.6.5 (default, May  7 2018, 10:07:07)
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

Does the error reproduce for you for these two examples or is there additional code that's needed for reproduction? Also tested with the unreleased 2.5.0 from the `master` branch and can't reproduce it there as well.

**Comment by same-id** [comment]
I think this now introduces a different bug

```
try:
  n = int(s)
  if n < 1:
    raise ValueError()
  return n
except ValueError:
  parser.error('parser error') # calls system exit
```

even if I replace parser.error() with an explicit function that is marked with `-> typing.NoReturn` as suggested in #4122 is still a false positive.

The only solution now is to preinitialize:

```
n = None
try:
  n = int(s)
  if n < 1:
    raise ValueError()
except ValueError:
  parser.error('parser error') # calls system exit
return n
```

Which isn't that great since it requires moving `n` to a less confined scope

**Comment by hippo91** [comment]
@same-id thanks for the report. I can reproduce. Can you open a new issue for this please?

**Comment by hippo91** [comment]
By the way another solution seems to add a `return` statement just before the `parser.error` call.

**Comment by hippo91** [comment]
@same-id i opened #4188

**Comment by same-id** [comment]
Thanks!
